core/risk/engine.go

```diff
@@ -554,3 +556,32 @@ func (m marginChange) Transfer() *types.Transfer {
 func (m marginChange) MarginLevels() *types.MarginLevels {
 	return m.margins
 }
+
+func orderReducesExposure(ord *types.Order, pos events.Margin) bool {
```

Comment:
We have this code (or something very similar to it) in the execution engine. If the order reduces exposure, we outright skip the margin check, so I'm not sure if we'd ever hit a case where this returns `true` (in execution/market.go, the `submitValidatedOrder` function):

```go
	pos := m.position.RegisterOrder(ctx, order)
	checkMargin := true
	if !isPegged && ok {
		oldVol, newVol := pos.Size()+pos.Buy()-pos.Sell(), oldPos.Size()+pos.Buy()-pos.Sell()
		if oldVol < 0 {
			oldVol = -oldVol
		}
		if newVol < 0 {
			newVol = -newVol
		}
		// check margin if the new volume is greater, or the same (implying long to short, or short to long)
		checkMargin = oldVol <= newVol
	}

	// Perform check and allocate margin unless the order is (partially) closing the party position
	if checkMargin {
		if err := m.checkMarginForOrder(ctx, pos, order); err != nil {
			if m.log.GetLevel() <= logging.DebugLevel {
				m.log.Debug("Unable to check/add margin for party",
					logging.Order(*order), logging.Error(err))
			}
			_ = m.unregisterAndReject(
				ctx, order, types.OrderErrorMarginCheckFailed)
			return nil, nil, ErrMarginCheckFailed
		}
```

Reply:
oh so it's still here, I thought it disappeared because I wasn't getting the behaviour I expected with the test.

Reply:
Ok so I moved the new method up and using it instead of the above now.